src/backend/src/services/organizations.service.ts

```diff
@@ -0,0 +1,75 @@
+import { User } from '@prisma/client';
+import { LinkCreateArgs, isAdmin } from 'shared';
+import prisma from '../prisma/prisma';
+import { AccessDeniedAdminOnlyException, HttpException } from '../utils/errors.utils';
+import { userHasPermission } from '../utils/users.utils';
+
+export default class OrganizationsService {
+  /**
+   * sets an organizations useful links
+   * @param submitter the user who is setting the links
+   * @param organizationId the organization which the links will be set up
+   * @param links the links which are being set
+   */
+  static async setUsefulLinks(submitter: User, organizationId: string, links: LinkCreateArgs[]) {
+    if (!(await userHasPermission(submitter.userId, organizationId, isAdmin)))
+      throw new AccessDeniedAdminOnlyException('update useful links');
+
+    const currentLinks = await prisma.organization.findUnique({
+      where: { organizationId },
+      select: { usefulLinks: { select: { linkId: true } } }
+    });
+
+    const currentLinkIds = currentLinks?.usefulLinks.map((link) => link.linkId);
+
+    await prisma.link.deleteMany({
+      where: {
+        linkId: { in: currentLinkIds }
+      }
+    });
+
+    for (const link of links) {
```

Comment:
have this be a map ad instead of updateing the organization each time just do an update at the end where you link all the linkIds

Reply:
id rather this be a helper function that return a list of linkIds that you use here